rules/TypeDeclaration/TypeInferer/SilentVoidResolver.php

```diff
@@ -48,7 +48,7 @@ public function hasExclusiveVoid(ClassMethod | Closure | Function_ $functionLike
             return false;
         }
 
-        if ($this->externalFullyQualifiedAnalyzer->hasExternalFullyQualifieds($classLike)) {
+        if ($classLike instanceof Stmt\ClassLike && $this->externalFullyQualifiedAnalyzer->hasExternalFullyQualifieds($classLike)) {
```

Comment:
the `Interface_` and `Trait_` already checked before, `Class_` only should be fine here.

```suggestion
        if ($classLike instanceof Stmt\Class_ && $this->externalFullyQualifiedAnalyzer->hasExternalFullyQualifieds($classLike)) {
```

Reply:
updated